lms/djangoapps/course_api/serializers.py

```diff
@@ -0,0 +1,79 @@
+"""
+Course API Serializers.  Representing course catalog data
+"""
+
+import urllib
+
+from django.core.urlresolvers import reverse
+from django.template import defaultfilters
+
+from rest_framework import serializers
+
+from xmodule.course_module import DEFAULT_START_DATE
+from lms.djangoapps.courseware.courses import (
+    course_image_url,
+    get_course_about_section,
+)
+
+
+class CourseSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer for Course objects
+    """
+
+    id = serializers.CharField(read_only=True)  # pylint: disable=invalid-name
+    name = serializers.CharField(source='display_name_with_default')
+    number = serializers.CharField(source='display_number_with_default')
+    org = serializers.CharField(source='display_org_with_default')
+    description = serializers.SerializerMethodField()
+    course_image = serializers.SerializerMethodField()
+    start = serializers.DateTimeField()
+    start_type = serializers.SerializerMethodField()
+    start_display = serializers.SerializerMethodField()
+    end = serializers.DateTimeField()
+    enrollment_start = serializers.DateTimeField()
+    enrollment_end = serializers.DateTimeField()
+    blocks_url = serializers.SerializerMethodField()
+
+    def get_start_type(self, course):
```

Comment:
When I talked to @nasthagiri, I pointed her to this wiki: https://openedx.atlassian.net/wiki/display/ECOM/Course+Discovery+API%3A+RESTful+Field+Specification

Those are the fields that we're intending to put into the Course Discovery api.

Ideally, I think this implementation of the course api would use the same fields and end-points, so that clients of this API can be easily ported over to the Course Discovery API when that is available.

The question of whether the courses coming from this API should be filtered by user or not is another that will make it harder to port to the Course Discovery API in the future.